### ISSUE TITLE
Controller crashes on switch disconnects

### ISSUE THREAD

**Issue by seliopou** [frenetic-lang/frenetic#374]
When a switch disconnects from the network, it brings the whole controller down with it and produces the following backtrace:

```
((pid 6198) (thread_id 0) (2014-12-09 15:59:29.715482-05:00)
 "unhandled exception in Async scheduler"
 ("unhandled exception"
  ((lib/monitor.ml.Error_
    ((exn Not_found)
     (backtrace
      ("Raised at file \"map.ml\", line 117, characters 16-25"
       "Called from file \"lib/Network.ml\", line 226, characters 10-40"
       "Called from file \"async/learning.ml\", line 75, characters 19-29"
       "Called from file \"async/Async_NetKAT.ml\", line 78, characters 8-18"
       "Called from file \"async/raw_app.ml\", line 132, characters 8-18"
       "Called from file \"async/raw_app.ml\", line 314, characters 20-31"
       "Called from file \"lib/deferred.ml\", line 145, characters 23-30"
       "Called from file \"lib/jobs.ml\", line 214, characters 10-13" ""))
     (monitor
      (((name main) (here ()) (id 1) (has_seen_error true)
        (is_detached false) (kill_index 0))))))
   (Pid 6198))))
```
This error is the result of the learning application querying the nib for a switch that is no longer there, indicating that the nib has been modified in response to an event before the applications have had the opportunity to process the event.

**Comment by arjunguha** [comment]
@reitblatt did you fix address this?

**Comment by reitblatt** [comment]
No, I don't think so.

**Comment by craig-riecke** [comment]
Fixed by Pull Request #416.
